pkg/reconciler/reconciler.go

```diff
@@ -0,0 +1,75 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type Reconciler interface {
+	Reconcile(ctx context.Context, c client.Client, scheme *runtime.Scheme) error
+}
+
+type UpdateReconciler struct {
+	controller metav1.Object
```

Comment:
I feel that `controller` is a bit overloaded.

```suggestion
	controlledBy metav1.Object
```